bulkredditdownloader/file_name_formatter.py

```diff
@@ -41,6 +42,10 @@ def _format_name(submission: praw.models.Submission, format_string: str) -> str:
                 logger.log(9, f'Found key string {key} in name')
 
         result = result.replace('/', '')
+
+        if platform.system() == 'Windows':
+            result = FileNameFormatter._format_for_windows(result)
+
```

Comment:
We should not format file names platform conditionally. Because, bdfr is commonly used in headless servers which run on Linux. However, when transfering files to a personal computer running Windows, they will cause problems.

On the other hand, right now, it is throwing file name issues because of Unicode characters (emojis, mainly).

So, I suggest replacing _**all**_ non-ASCII characters in addition to the ones that are problematic to Windows.

Reply:
I disagree, there's no reason to discard information because of the poor design of Windows and the archaic filesystem that it uses. There are many Linux tools to clean file names to be compatible with Windows, and many implementations of SAMBA and otherfile sharing protocols from Linux to Windows, include this functionality. When emojis and regular punctuation are used in the titles of posts, the BDFR should be able to accurately save this information unless doing so will crash the program.

Reply:
Emojis crash program on Windows.